collectives/static/js/technician.js

```diff
@@ -0,0 +1,143 @@
+          columns:[
+            {field:"name",  formatter:"link",  title:"Nom", headerFilter:true, formatterParams:{
+                labelField :"name",
+                urlPrefix:"https://doc.collectives.cafannecy.fr/configuration.html#config.",
+                }},
+            {field:"content", title:"Valeur", headerFilter: "input", formatter: "textarea", 
+                    editor: true, cellEdited: updateConf, 
+                    mutatorData :  (data) => data.replaceAll('\\n', '\n') 
+                },
+         ],
+
+            locale:true,
+            langs:{
+                "fr-fr":{
+                    "ajax":{
+                        "loading":"Chargement", //ajax loader text
+                        "error":"Erreur", //ajax error text
+                    },
+                    "pagination":{
+                        "page_size":"Configuration par page", //label for the page size select element
+                        "first":"Début", //text for the first page button
+                        "first_title":"Première Page", //tooltip text for the first page button
+                        "last":"Fin",
+                        "last_title":"Dernière Page",
+                        "prev":"Précédente",
+                        "prev_title":"Page Précédente",
+                        "next":"Suivante",
+                        "next_title":"Page Suivante",
+                    }
+                }
+            },
+    });
+}
+
+
+// Check then upload the modified configuration item into server
+// Checks are rudimentary.
+function updateConf(cell){
+    var data = cell.getRow().getData();
+    console.log(data['content']);
+    data['content'] = patchNewLine(data['content']);
+
+    if(! isJsonString(data['content'])){
+        alert('La valeur n\'est pas correcte: ce n\'est pas une chaine JSON. Veuillez la corriger');
+        console.error("Wrong JSON:")
+        console.error(data['content'])
+        return false;
+    }
+
+    if(! confirm(`Vous allez mettre à jour ${data['name']}. Confirmez-vous?`)){
+        return false;
+    }
+
+    xhr = new XMLHttpRequest();
+    xhr.open('POST', '/api/administration/configuration');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.setRequestHeader('X-CSRFToken', csrf_token);
+    xhr.onload = function() {
+        if (xhr.status === 200) {
+            cell.getTable().replaceData();
+        }
+        else {
+            cell.getTable().replaceData();
+            alert('Request failed.  Returned status of ' + xhr.status);
+        }
+    };
+    xhr.send(JSON.stringify(data));
+    
+}
+
+// Check if the string is a valid JSON dump
+function isJsonString(str) {
+    try {
+        JSON.parse(str);
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
+
+/* Function to convert newLines into \n in JSON strings.
+It does not convert new line outside JSON strings */
+function patchNewLine(dataString){
+    var data = Array.from(dataString);
+
```

Comment:
Cela ne me parait pas très robuste, par exemple je n'ai pas l'impression que ca prenne en compute les choses comme `" \" "`. Je suggérais d'utiliser un vrai parser JSON, itérer sur les attributes, escaper les string puis ré-encoder le JSON en string pour l'affichage. Plus lent, mais au moins ca prend en compte tous les cas particuliers.